### DISCUSSION TITLE
Is there a way to route midi to an LV2 _effect_?

### DISCUSSION THREAD

**Discussion by io7m** [muse-sequencer/muse#1281] [Q&A] [answered]
Hello!

I have a few LV2 plugins that are used as effects but that also have MIDI/event inputs. A good example of this is the "Example MIDI Gate" plugin:

https://github.com/lv2/lv2/tree/master/plugins/eg-midigate.lv2

It's an effect that essentially only passes an audio signal when there's a MIDI note `ON`, on the associated MIDI input.

I can load this plugin as an effect on an audio channel in Muse, but I can't work out how to route any MIDI to it. Is this actually possible?

**Comment by terminator356** [comment, marked as answer]
Thanks for your question. Unfortunately we have no mechanism to route midi to the effect rack and its effects.
Nor from effect to effect in a rack, which might be a logical addition. 
(We also lack, for example, mid track to midi track routing. I think someone had a clumsy /dev/midi* user-workaround once.)

The usual workaround is to try to use the plugin as a synth track which supports midi, and use say, a group track to act as a 
 'pre-rack' to 'split' the effect rack right where your midi effect needs to be, and route audio from the group track to the synth track,
 and put the remaining effects from the 'split' into the synth track's effect rack. Thus acting like one continuous effect rack,
 including your midi plugin. Something like that.

I have thought about how (and whether) to route midi to the effect rack many times. 
It's still a concept. Questions like how should the user interface look and where should it be - the routing 
 popup menus and/or the 'advanced' router? How to choose ports, channels etc. How to provide routing services, and so on.
Being some years since last considering these things, maybe another attempt could lay things out more clearly, 
at least conceptually. Naturally it's complicated. If it's me, it might be some time before I could get to it.

**Comment by io7m** [comment]
Thanks for the info!

To be honest, I think the use case is minor enough that it's not worth putting in any serious work for it. It's good to know that I wasn't just missing something obvious. :slightly_smiling_face:
